manifests/repo/dnfmodule.pp

```diff
@@ -0,0 +1,23 @@
+# @summary Manage the DNF module
+#
+# On EL8+ DNF can manage modules.
+# This is a method of providing multiple versions on the same OS.
+# Only one DNF module stream can be active at the same time.
+#
+# @param ensure
+#   Value of ensure parameter for nodejs dnf module package
+#
+# @param module
+#   Name of the nodejs dnf package
+#
+# @api private
+class nodejs::repo::dnfmodule (
+  String[1] $ensure = $nodejs::repo_version,
+  String[1] $module = 'nodejs',
+) {
+  package { 'nodejs dnf module':
```

Comment:
Should we add a `fail()` for non RH family or major ≠ 8/9?

Reply:
Looking at `nodejs::repo::nodesource` it'd be consistent, though technically it also works on Fedora. At least Fedora 38 still has a NodeJS 16 module (while 18 non-modular is the default).

Given people who look for modularity know what it is, I wouldn't be too strict about it.